Library/Formula/llvm.rb

```diff
@@ -116,18 +142,73 @@ def install
     # Apple's libstdc++ is too old to build LLVM
     ENV.libcxx if ENV.compiler == :clang
 
-    if build.with?("lldb") && build.without?("clang")
-      raise "Building LLDB needs Clang support library."
+    if build.with? "clang"
+      (buildpath/"tools/clang").install resource("clang")
+    end
+
+    if build.with? "extra-tools"
+      if build.with? "clang"
+        (buildpath/"tools/clang/tools/extra").install resource("extra-tools")
+      else
+        odie "clang's extra tools must be built with clang"
+      end
     end
 
-    if build.with? "clang"
+    if build.with? "libcxx"
       (buildpath/"projects/libcxx").install resource("libcxx")
-      (buildpath/"tools/clang").install resource("clang")
-      (buildpath/"tools/clang/tools/extra").install resource("clang-tools-extra")
     end
 
-    (buildpath/"tools/lld").install resource("lld") if build.with? "lld"
-    (buildpath/"tools/lldb").install resource("lldb") if build.with? "lldb"
+    if build.with? "lld"
+      (buildpath/"tools/lld").install resource("lld")
+    end
+
+    if build.with? "lldb"
+      if build.with? "clang"
+        (buildpath/"tools/lldb").install resource("lldb")
+
+        # Building lldb requires a code signing certificate.
+        # The instructions provided by llvm creates this certificate in the
+        # user's login keychain. Unfortunately, the login keychain is not in
+        # the search path in a superenv build. The following three lines add
+        # the login keychain to ~/Library/Preferences/com.apple.security.plist,
+        # which adds it to the superenv keychain search path.
+        mkdir_p "#{ENV["HOME"]}/Library/Preferences"
+        username = `whoami`.strip! # Has ending newline otherwise
+        system "security", "list-keychains", "-d", "user", "-s", "/Users/#{username}/Library/Keychains/login.keychain"
```

Comment:
You can use `ENV["USER"]` instead of `whoami`. Also can you test the option locally with sandbox enabled, as `brew install llvm --with-clang --with-lldb --sandbox -v`. I want to make sure it doesn't write file to place not permitted. Thanks.